src/porepy/numerics/ad/operators.py

```diff
@@ -1853,7 +1865,7 @@ def __init__(self, variables: list[Variable]) -> None:
                 len(set(time_indices)) == 1
             ), "Cannot create md-variable from variables at different time steps."
             assert (
-                len(set(iterate_indices)) == 1
+                len(set(iter_indices)) == 1 and len(set(current_iter)) == 1
```

Comment:
Why the check of current_iter?

Reply:
Thanks for the explanation. Could you please add one line of comment.

Reply:
Documentation added.